### PR TITLE
Filter spatial

### PR DESCRIPTION
**Pull request by masawdah**
Hi team, 
together with @clausmichele have implemented spatial_filter process and added a test for it.

### CODE REVIEW

**Comment by clausmichele** [comment]
This PR includes implementations of filter_spatial and mask_polygon. We tested them on full S2 tiles, you also may need to test them on the EODC back-end and report if you see some issues.

Current implementation supports only geoJSON vector-cubes, since currently the `load_vector_cube` process has not been implemted and we didn't discuss on which data structure to use for it (probably dask-geopandas?)

**Comment by clausmichele** [comment]
@ValentinaHutter @GeraldIr can you start reviewing this PR?

File: openeo_processes_dask/process_implementations/cubes/_filter.py
Comment (GeraldIr) [inline openeo_processes_dask/process_implementations/cubes/_filter.py:1]: It seems that the dimensions defined at the beginning of the filter_spatial function are unused. I'm unsure if they need to be used or if it's leftover code, so i thought I'd mention it as these values are used in the other functions.

**Review by GeraldIr — COMMENTED** [review]
Just the one question about unused variables, but that doesn't seem to impact the logic of the function, otherwise LGTM!

File: openeo_processes_dask/process_implementations/cubes/_filter.py
Comment (masawdah) [reply openeo_processes_dask/process_implementations/cubes/_filter.py:1]: Indeed they're leftover code. I'll remove them.

**Comment by clausmichele** [comment]
@GeraldIr @ValentinaHutter I think it would be good to test the implementation at EODC with your cluster and with your data, to check how it behaves with large datasets.

**Pull request merged**